src/runtime/crt/memory/stack_allocator.c

```diff
@@ -79,8 +79,14 @@ tvm_crt_error_t StackMemoryManager_Free(tvm_workspace_t* tvm_runtime_workspace,
 
 tvm_crt_error_t StackMemoryManager_Init(tvm_workspace_t* tvm_runtime_workspace,
                                         uint8_t* g_aot_memory, size_t workspace_size) {
+  // We need to round up g_aot_memory in case it is not aligned to
+  // TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES.
+  uintptr_t unaligned_mask = TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES - 1;
+  uint32_t offset = TVM_RUNTIME_ALLOC_ALIGNMENT_BYTES - ((uintptr_t)g_aot_memory & unaligned_mask);
```

Comment:
it might be easier to understand this computation by subtracting the passed-in `g_aot_memory` from the newly computed one below.

Reply:
I agree, I changed it to that.